### ISSUE TITLE
Remove 'Update EPUB Cover?' Download Option from FFF Plugin

### ISSUE THREAD

**Issue by JimmXinu** [JimmXinu/FanFicFare#878]
### TL;DR
The [FanFicFare Cover Settings](../wiki/Cover-Settings-Calibre-Plugin) wiki page documents using covers in FanFicFare Plugin after the changes made in FanFicFare v4.16.

FanFicFare v4.16 adds a _Set Calibre Cover Only for New Books_ checkbox in the plugin config Calibre Cover tag plus `use_old_cover` and `force_cover_image` `personal.ini` settings to be cleaner replacements for **Update EPUB Cover?** and `never_make_cover`.

You can get the same behavior as unchecked **Update EPUB Cover?** with `personal.ini` setting:
```
[overrides]
never_make_cover:true
```
But you might find a better idea in [FanFicFare Cover Settings](../wiki/Cover-Settings-Calibre-Plugin).

### The 'Update EPUB Cover?' Checkbox, a History
Since practically day one of the FanFicFare Plugin, every download has had a 'Download Option' checkbox labeled **Update EPUB Cover?**  (And corresponding **Default Update EPUB Cover when Updating EPUB?** setting in FFF Config.)

This was for the benefit of early users who wanted to use the plugin, but only wanted to download covers _some_ of the time.  They were vocal enough about FFF overwriting covers at the time that **Default Update EPUB Cover when Updating EPUB?** has always defaulted to unchecked for new users.  

In their defense, the first generation of FFF plugin users often had existing libraries of fanfics and FFF (then FFDL) overwriting their  hand crafted covers was rude, at best.  And, at the time, FFF shipped with `include_images:false` set by default, so downloading images at all was a bit novel.

### How it worked:
The way **Update EPUB Cover?** worked, was when it was **UNchecked** FFF literally appended an extra setting to `personal.ini` when it was processed:
```
[overrides]
never_make_cover: true
```

In retrospect, a Plugin setting that changed `personal.ini` was a mistake.

`never_make_cover: true` causes the FanFicFare downloader to *not* use a story's specific cover images (if there is one) and suppresses any cover that may have resulted from `make_firstimage_cover:true`.  It does *not* prevent any `default_cover_image` that may be set, nor the inclusion of an existing cover image during epub update.

What **Update EPUB Cover?** did not do (directly) was control whether *Calibre's* cover is updated.  This has long been a point of confusion, especially with the later addition of the 'Calibre Cover' tab in FFF's Config.

### Removal of a Feature
In FanFicFare v4.16, the 'Download Option' **Update EPUB Cover?**  checkbox is disabled.  The corresponding **Default Update EPUB Cover when Updating EPUB?** setting in FFF Config still works, but a warning dialog is displayed any time **Default Update EPUB Cover when Updating EPUB?** will be used or when it is unset.

**Update EPUB Cover?** and **Default Update EPUB Cover when Updating EPUB?** will be removed completely in a future release.

At the same, the 'new install' default values for several other Calibre Cover features are changed to be more useful and intuitive for new users and some new features added, namely: 
- **Set Calibre Cover Only for New Books** in FFF Config 'Calibre Cover' tab;
- `use_old_cover` in `personal.ini`, and;
- `force_cover_image` in `personal.ini`. 

--------
### FanFicFare CLI `--update-cover` option

The FFF CLI's `--update-cover` option is analogous to **Update EPUB Cover?**, eg, applying `never_make_cover: true` when `--update-cover` is *not* specified.  

But while **Update EPUB Cover?** was applied all the time, including to new downloads, the absence of `--update-cover` in CLI only applies `never_make_cover: true` when *actually* updating an existing epub.

That, and the tendency for Calibre Plugin users to care about covers significantly more, means the CLI `--update-cover` option is not being changed at this time.  But `use_old_cover` and `force_cover_image` in `personal.ini` will be available for CLI.

**Comment by JimmXinu** [comment]
**Update EPUB Cover?** is removed entirely in test version v4.17.1 and is planned to not appear in release v4.18.

**Comment by JimmXinu** [comment]
Closing this issue with the release of v4.18.0.
